up-to-date description of the Horace installation on IDAAAS … (#1290)
pace-neutrons/Horace: documentation/smg/11_ISIS_release.md

# ISIS Release Process

> Horace deployed on iDAAaS system and machines dedicated for excitations as a single git repository available 

Reviewer: ```suggestion
Horace is deployed on iDAAaS system and machines dedicated for Excitations as a single git repository available 
```

> read-only for excitation machine users and providing read/write access for the members of the excitations group.

Reviewer: ```suggestion
read-only for Excitations machine users and providing read/write access for the members of the Excitations group.
```

> As the single place of the Horace code is available to all machines, the changes to the code made by releasers are immediately available to all iDAAaS machines.

Reviewer: ```suggestion
As the single place where the Horace code is available to all machines, the changes to the code made by releasers are immediately available to all iDAAaS machines.
```

## Horace installation in ISIS

> Common path `/usr/local/mprogs/Users` is added to the search path of all MATLAB sessions available to users processing the results of inelastic neutron scattering experiments in ISIS.

Reviewer: ```suggestion
A common filepath `/usr/local/mprogs/Users` is added to the search path of all MATLAB sessions available to users processing the results of inelastic neutron scattering experiments in ISIS.
```

> On the machines, managed by excitation group this is achieved by adding this path to the end of the `{MATLAB Installation Folder}/toolbox/local/pathdef.m` file, though recent iDAAaS virtual machines seems use different approach for adding this path to the MATLAB search path.

Reviewer: ```suggestion
On the machines managed by the Excitations group, this is achieved by adding this path to the end of the list of paths referenced in the`{MATLAB Installation Folder}/toolbox/local/pathdef.m` file. 

Note that recent iDAaaS virtual machines use a different approach for adding this path to the MATLAB search path.
```

There needs to be more detail about the different approach. We can't have an instruction that doesn't work on half the IDAaaS machines.

Author: I hondestly do not know what they do but `pathdef.m` is unchanged. Probably they wrap Matlab launch script into into some other script or set up some enviromental variables.

> This path contains Horace initialization script `horace_on.m` which, on execution, adds all necessary folders to MATLAB search path and initialize all other Horace variables necessary for Horace operations. (i.e. Horace configurations). This folder also contains the worker script, which invokes `horace_on` script and initializes Horace for all parallel MATLAB sessions. 

Reviewer: ```suggestion
This path includes the Horace initialization script `horace_on.m` which, on execution, adds all necessary folders to MATLAB search path and initializes all other Horace variables necessary for Horace operations. (i.e. Horace configurations). This folder also contains the worker script which invokes the `horace_on` script and initializes Horace for all parallel MATLAB sessions. 
```
Did you mean i.e. so only configurations are initialized or did you mean e.g. so other things are initialized as well?

Author: should be e.g. indeed.

Administrative script (Horace developers do not have write access to it) `/var/daaas/dynamic/matlabsetup.m` is executed on start up of MATLAB user sessions of iDAAaS machines and invokes `horace_on` for user sessions dedicated for using Horace. 

## Deploy Location
The repositories with the Horace code on iDAAaS file system are currently located under:
`/mnt/ceph/auxiliary/excitations/isis_direct_soft/` 

> Number of the actual repositories are cloned under this location:

Reviewer: ```suggestion
A number of the actual repositories are cloned under this location:
```

The following repositories are currently cloned there:

 -- `Horace_git` -- the repository prepared for recent Horace release. As we have not actually released Horace, `master` is currently checked out there. Horace-4 master have also `Herbert` code included.
 
 -- `Horace_3_6` -- the repository for last release of Horace-3. `horace_3_6_3` branch containing latest bugfix release of   Horace.
 
 -- `Horace_3_2` -- historical Horace release hopefully not used any more
 
 -- `Horace_bugfix` -- the special repository clone, with purpose, described later.
 

>  As previous to version 4 versions of Horace needed correspondent versions of Herbert, three clones of Herbert repositories, corresponding to appropriate Horace repositories are also cloned here, namely: `Herbert_git`, `Herbert_3_6`, `Herbert_3_2`, and `Herbert_bugfix`

Reviewer: ```suggestion
 As, previous to version 4, versions of Horace needed correspondent versions of Herbert, three clones of Herbert repositories, corresponding to appropriate Horace repositories are also cloned here, namely: `Herbert_git`, `Herbert_3_6`, `Herbert_3_2`, and `Herbert_bugfix`
```

`/usr/local/mprogs/Herbert_bugfix` -> `/home/isis_direct_soft/Herbert_bugfix`
`/usr/local/mprogs/Horace_bugfix` -> `/home/isis_direct_soft/Horace_bugfix`

> Recent Horace version does not need separate Horace, so only second link should be used for running bugfix code.

Reviewer: ```suggestion
Recent Horace version does not need separate Herbert, so only second link should be used for running bugfix code.
```

The code is invoked by calling `horace_on` with the path to `Horace_bugfix` folder, i.e: `horace_on("/usr/local/mprogs/Horace_bugfix")`

## Deploy Process

As soon as release branch is tagged and created in the repository, manual build/release process is preformed in the private area of iDaaaS machine to obtain release artefacts for the iDAAaS computational service. It is also possible to pull the release artefacts from Jenkins builds, though this process needs thorough testing. 

> The extraction of the release artefacts will be automated in a future, as [iDaaaS testing pipeline](https://github.com/pace-neutrons/Horace/issues/271) is implemented.

Reviewer: ```suggestion
The extraction of the release artefacts will be automated in future, as [iDaaaS testing pipeline](https://github.com/pace-neutrons/Horace/issues/271) is implemented.
```

> The releaser then manually checks out the release branch into `Horace_git` areas above and copies the release artefacts into appropriate places of the code tree.

Reviewer: I am finding this unclear. The exact processes to be used by the releaser to do this manual checkout and copy needs to be specified in detail.

Author: added more about the release artefacts though they are descrined in other document

> Minor changes and bugfixes do not need the rebuild of the release artefacts and performed by pulling recent merges to the release branch into `Horace_git` repository.

Reviewer: you don't say WHAT is performed. The verb needs a subject.

Author: hopefully clarified

## Horace-4 Transitional period

> While Horace-4 is undergoing pre-release process, it is expected to be used alongside with previous version of Horace. This is why Horace master branch is currently checked out at `Horace_git` path (recent Horace is combined with Herbert) and latest (3.6.3) version of Horace/Herbert is checked out at `Horace_3_6` and `Herbert_git` path-es. 

Reviewer: ```suggestion
While Horace-4 is undergoing pre-release process, it is expected to be used alongside previous version of Horace. This is why Horace master branch is currently checked out at `Horace_git` path (recent Horace is combined with Herbert) and latest (3.6.3) version of Horace/Herbert is checked out at `Horace_3_6` and `Herbert_git` paths. 
```

Two additional versions of `horace_on` script, namely `horace3_on` and `horace4_on` scripts are available in `/usr/local/mprogs/Users` folder and allow user to switch between different versions of Horace calling appropriate script. `horace_on` script called on startup of MATLAB sessions is currently the direct copy of `horace3_on` script.  It will become the copy of `horace4_on` script when Horace-4 will become the default version of Horace.

## Notes

>  The the group which have write access to the Horace repositories is named `standarduser`, and IDAAaS support is responsible for adding users to this group. Contact supportanalysis@stfc.ac.uk to be included in this group and obtain write access to the repositories. Normal users have read-only access to the code area, sufficient to use the code in their computations.

Reviewer: ```suggestion
 The linux file-group which has write access to the Horace repositories is named `standarduser`, and IDAaaS support is responsible for adding users to this group. Contact supportanalysis@stfc.ac.uk to be included in this group and obtain write access to the repositories. Normal users have read-only access to the code area, sufficient to use the code in their computations.
```

#### Contact

iDaaaS support team : support@analysis.stfc.ac.uk

## SCARF

>  Horace is currently not available on SCARF and this chapter is not verified and probably incorrect. This was working in the past and will probably be organized in some similar way in a future. 

Reviewer: ```suggestion
 Horace is currently not available on SCARF and the remainder of this chapter is the previously existing text in the file. It is hence not verified and probably incorrect. This was working in the past and will probably be organized in some similar way in a future. 
```
